angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete.component.ts

```diff
@@ -0,0 +1,421 @@
+          this.input.ref.nativeElement.parentNode,
+          'aria-expanded',
+          'true',
+        );
+      }
+
+      this.openedEvent.emit();
+      this.isOpen = true;
+    }
+  }
+
+  /**
+   * @ignore
+   * Track the index of the currently-highlighted Autocomplete result item.
+   */
+  highlightedIndex = -1;
+
+  /**
+   * @ignore
+   */
+  ngAfterContentInit(): void {
+    if (this.results) {
+      if (this.isOpen) {
+        this.showResults();
+      } else {
+        this.hideResults();
+      }
+
+      if (this.input) {
+        // set aria-controls on the input to the id of the results
+        this.generateAriaControls(
+          this.input.ref.nativeElement,
+          this.results.nativeElement,
+        );
+
+        // set aria-owns on the input's parent to the id of the results
+        this.generateAriaOwns(
+          this.input.ref.nativeElement.parentNode,
+          this.results.nativeElement,
+        );
+      }
+    }
+
+    // if itemSelectedEvent is specified, also set that as the click event on each result
+    if (this.itemSelectedEvent) {
+      this.resultItems.forEach((element) => {
+        element.clickedEvent = this.itemSelectedEvent;
+      });
+    }
+  }
+
+  /**
+   * @ignore
+   * This logic needs to happen in this event so that the input width is
+   * correctly calculated.
+   */
+  ngAfterViewInit(): void {
+    if (this.input) {
+      this.calculateResultsWidth();
+    }
+  }
+
+  /**
+   * @ignore
+   * Set the max-width on the Autocomplete results equal to the current
+   * width of the input element.
+   */
+  calculateResultsWidth() {
+    if (this.input) {
+      const currentInputWidth = this.input.ref.nativeElement.offsetWidth;
+
+      if (this.results) {
+        this.renderer.setAttribute(
+          this.results.nativeElement,
+          'style',
+          'max-width:' + currentInputWidth + 'px',
+        );
+      }
+    }
+  }
+  /** @ignore */
+  isUpPressed = (e) => e.key === 'ArrowUp' || e.keyCode === 38;
+  /** @ignore */
+  isDownPressed = (e) => e.key === 'ArrowDown' || e.keyCode === 40;
+  /** @ignore */
+  isEnterPressed = (e) => e.key === 'Enter' || e.keyCode === 13;
+  /** @ignore */
+  isEscapePressed = (e) =>
+    e.key === 'Escape' || e.key === 'Esc' || e.keyCode === 27;
+
+  /**
+   * @ignore
+   * Copy the value of the id attribute on contentElement
+   * into the aria-controls attribute on triggerElement.
+   * Generate a unique ID if needed.
+   * @param triggerElement The element that will receive the aria-controls attribute.
+   * @param contentElement The element that will receive an id if needed.
+   */
+  generateAriaControls = (triggerElement, contentElement) => {
+    const triggerAriaControls = triggerElement.getAttribute('aria-controls');
+    let contentId = contentElement.getAttribute('id');
+
+    // Warn if aria-controls exists but the id does not
+    if (triggerAriaControls && !contentId) {
+      console.warn(
```

Comment:
I know we've talked about this a few times. My recollection is that we decided not to do warnings like this moving forward because we didn't want errors/warnings in the console of prod applications. I'd like to double check where we landed on that decision.

Reply:
I discussed this with @afebbraro in the team chat. The consensus so far is that Spark should not try to recreate functionality in the Axe browser extension. Spark should generate IDs and aria values and apply them when possible, but if the implementing engineer passes in invalid ids we're going to just leave them alone and trust Axe to flag the error.

If this sounds good to everyone, I'll change this PR so `generateAriaOwns` and `generateAriaControls` do not console.warn.